docs/source/howto/run_containerized_codes.rst

```diff
@@ -0,0 +1,160 @@
+
+
+.. note::
+
+   For the purposes of this how-to guide, we will be using `Singularity <https://singularity-docs.readthedocs.io/en/latest/>`__ as the containerizing technology.
+   Singularity is easy to install and designed to support scientific computing on HPC compute resources.
+   There are other similar containerizing technologies for HPC, such as `Sarus <https://sarus.readthedocs.io/en/stable/>`__ and `Shifter <https://github.com/NERSC/shifter>`__ which are equally supported by the containerized code plugins, although only Sarus and Singularity have been explicitly tested.
+
+.. _how-to:run-containerized_codes:installed:
+
+Installed containerized code
+============================
+
+The :class:`aiida.orm.nodes.data.code.containerized.InstalledContainerizedCode` data plugin allows running a code inside a container that is available on a computer that is configured in AiiDA through the :class:`~aiida.orm.computers.Computer` class.
+Consider the example where we have a container ``docker://alpine`` installed on the ``localhost`` computer (if image not exist, it will be download automatically the first time this containerized code run), and we want to run the ``core.arithmetic.add`` calculation job plugin using the ``/bin/sh`` executable inside the container using ``Singularity``.
+To configure this containerized code, create the following configuration file:
+
+    .. code-block:: yaml
+
+        ---
+        label: add-singularity
+        description: add executed on singularity
+        default_calc_job_plugin: core.arithmetic.add
+        computer: localhost
+        filepath_executable: /bin/sh
+        image_name: docker://alpine
+        engine_command: singularity exec --bind $PWD:$PWD {image_name}
+        prepend_text: ' '
+        append_text: ' '
+
+Create the code by executing:
+
+.. code-block:: console
+
+    $ verdi code create core.code.installed.containerized --config singularity-add.yaml
+
+The created code can then be used to run a calculation just like any other code:
+
+.. code-block:: python
+
+    from aiida.engine import run_get_node
+    code = load_code('add-singularity@localhost')
+    builder = code.get_builder()
+    builder.x = Int(1)
+    builder.y = Int(2)
+    results, node = run_get_node(builder)
+
+The big advantage over the non-containerized :class:`aiida.orm.nodes.data.code.containerized.InstalledContainerizedCode` equivalent is that with the containerized version there is no need to explicitly install or compile the code on the remote machine.
+Note that this example requires Singularity to be installed on the target computer and that the ``filepath_executable`` is already part of the container image.
+
+.. note::
+    The first time that the containerized code is run, the image will be automatically pulled from Docker Hub, which may takes a while if the image is large.
```

Comment:
Although this may seem trivial, this is very important information and needs to be mentioned early